### PR TITLE
Update lint tooling

### PR DESCRIPTION
**Pull request by anandaroop**
This is a tooling update for Rosalind, before we start up feature work again.

When this project was spun up 2 yrs ago we [added](https://github.com/artsy/rosalind/pull/14) StandardJS. But it's more typical now for projects to use ESLint/TSLint and Prettier. Many projects also adopt automated pre-commit linting.

This PR:
- migrates us from StandardJS to ESLint + Prettier (and makes them play nice)
- Introduces lint-staged for pre-commit linting of JS _and_ Ruby
- Lints all the things

(Along the way we drop a git hook template, which is no longer needed thanks to lint-staged.)

One thing this PR doesn't do is try out [prettier-ruby](https://github.com/prettier/plugin-ruby). I'm interested in that, but probably in another PR.

TODO

- [x] Ensure CI checks still work (for those rascals who commit with `--no-verify`)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,7 +58,7 @@
     "setupTestFrameworkScriptFile": "<rootDir>/app/javascript/test/setup.js"
   },
   "scripts": {
-    "lint": "standard $(find config webpack -name *.js)",
+    "lint": "eslint $(find app/javascript -name *.js -or -name *.jsx)",
```
Comment (damassi) [inline package.json]: Maybe `eslint app/javascript --ext js,jsx`?

File: package.json [hunk 2]
```diff
@@ -78,7 +76,41 @@
...
+  },
+  "lint-staged": {
+    "app/javascript/**/*.{js,jsx}": [
+      "eslint --fix",
+      "git add"
+    ],
+    "**/*.rb": [
+      "bundle exec rubocop -a",
+      "git add"
+    ]
```
Comment (damassi) [inline package.json]: It's always a joy to see this set of functionality added to a codebase 👍

File: package.json (hunk 1 above)
Comment (anandaroop) [reply package.json]: Nice, didn't know about `--ext`. Fixed in f830f3d

**Comment by anandaroop** [comment]
Gonna merge, and see if #186 builds now

**Pull request merged**